File: 17Simpson/simpson.m
function resutlado = simpson(f,a,b)
  pkg load symbolic
  syms x
  h = (b-a)/2;
  
  x = a;
  fx0 = eval(f);
  x= (a+b)/2;
  fx1 = eval(f);
  
  x = b;
  
  fx2 = eval(f);
  
  resultado = (h/3)*(fx0 + 4*fx1 + fx2)
  return
endfunction

File: 17Simpson/simpson.py
import numpy as np
import sympy as sy
from sympy import symbols

def simpson(f,a,b):
    x = symbols('x')
    f = sy.sympify(f)
    h = (b-a)/2
    x1 = (a+b)/2

    fx0 = f.evalf(subs={x:a})
    fx1 = f.evalf(subs={x:x1})
    fx2 = f.evalf(subs={x:b})
    
    resultado = (h/3)*(fx0 + 4*fx1 + fx2)
    print(resultado)
    return resultado

simpson('ln(x)',2,5)
